torch/__init__.py

```diff
@@ -1135,14 +1135,32 @@ def get_default_device() -> "torch.device":
     r"""Gets the default ``torch.Tensor`` to be allocated on ``device``"""
     global _GLOBAL_DEVICE_CONTEXT
 
-    if hasattr(_GLOBAL_DEVICE_CONTEXT, "device_context"):
-        device = _GLOBAL_DEVICE_CONTEXT.device_context.device
+    from torch.overrides import _get_current_function_mode_stack
+    from torch.utils._device import DeviceContext
```

Comment:
I'm confused now what's the difference between torch._GLOBAL_DEVICE_CONTEXT and torch.utils._device.CURRENT_DEVICE ?
And if we change all access to respect the later, should we just delete the first one?

Reply:
Currently, I think `_GLOBAL_DEVICE_CONTEXT` is required to correctly implement the `set_default_device` with `DeviceContext` (it holds the DeviceContext object on which __enter__ and __exit__ is called)

https://github.com/pytorch/pytorch/blob/ece16584184a072ba0349ef57604867ed212e629/torch/__init__.py#L1205-L1218

Actually looking at [_device.py](https://github.com/pytorch/pytorch/blob/main/torch/utils/_device.py), I am not sure if `CURRENT_DEVICE` is needed for implementation.

But I think this (or above) refactoring can be addressed in a separate PR, wdyt?

Reply:
Gentle ping :)